Source/missiles.cpp

```diff
@@ -2904,23 +2900,24 @@ void AddNova(int mi, int sx, int sy, int dx, int dy, int midir, int8_t mienemy,
 
 void AddBlodboil(int mi, int sx, int sy, int dx, int dy, int midir, int8_t mienemy, int id, int dam)
 {
-	if (id == -1 || plr[id]._pSpellFlags & 6 || plr[id]._pHitPoints <= plr[id]._pLevel << 6) {
+	auto &player = plr[id];
+
+	if (id == -1 || player._pSpellFlags & 6 || player._pHitPoints <= player._pLevel << 6) {

```

Comment:
We should probably check for `id == -1` before attempting to access `plr[id]`.

Reply:
Looks like this one also cannot happen as blood boil is only cast by the player (`PM_DoSpell()`) and so always have an id.